Userland/Libraries/LibArchive/Tar.h

```diff
@@ -95,7 +95,17 @@ class [[gnu::packed]] TarFileHeader {
     // FIXME: support 2001-star size encoding
     size_t size() const { return get_field_as_integral(m_size); }
     time_t timestamp() const { return get_field_as_integral(m_timestamp); }
-    unsigned checksum() const { return get_field_as_integral(m_checksum); }
+    ErrorOr<unsigned> checksum() const
+    {
+        for (char character : m_checksum) {
+            if (character == 0 || character == ' ')
+                break;
+            if (character < '0' || character > '7')
+                return Error::from_errno(EINVAL);
+        }
```

Comment:
Why shouldn't this apply to all the fields that use `get_field_as_integral`? As far as I can tell all of them are using octals, and all of them would trigger the same `VERIFY` if non-octal values are found.